### ISSUE TITLE
Random seed for reproducibility

### ISSUE THREAD

**Issue by nesnoj** [openego/eGon-data#351]
The random module is used in several tasks. To make a further step towards reproducibility, a predefined seed should be used.
In a linear program it's sufficient to set the seed only once according to [this SO post](https://stackoverflow.com/a/11527011). But as we're running multiple tasks in parallel, the order of the drawings is not deterministic, right?

So what about defining a seed centrally in egon-data and set it in every affected script, does this sound like a good solution to you? @gnn What would be a good place to store the seed? I couldn't find an appropriate file, maybe datasets.yml but this is not the actual intention of the file.. (apart from the fact that we wanted to switch to a more decentralized concept here)

**Comment by nesnoj** [comment]
@gnn Could you please comment on this issue?
@AmeliaNadal @IlkaCu @ClaraBuettner Doyou make use of random in your scripts?

**Comment by IlkaCu** [comment]
I don't use random in my scripts, but I am pretty sure that @ClaraBuettner and maybe some of our students use it.

**Comment by nesnoj** [comment]
The following tasks seem to be affected:
- [ ] HH demand profiles #260 
- [ ] SocietyPrognosis
- [ ] small chp - there's at least a seed set, should be set to global seed

If you make use of a separate multiprocessing in a task, you should set the seed within the appropriate function that's called by the mp to make sure the drawing in the tasks do not interfere.

As soon as #409 is merged, the tasks above can be adjusted.

**Comment by gnn** [comment]
> So what about defining a seed centrally in egon-data and set it in every affected script, does this sound like a good solution to you? @gnn What would be a good place to store the seed? I couldn't find an appropriate file, maybe datasets.yml but this is not the actual intention of the file.. (apart from the fact that we wanted to switch to a more decentralized concept here)
I'd make that a command line parameter without a default. That way one can set it via `egon-data.configuration.yaml` in the working directory and it would be easily available via the dictionary returned by `config.settings()`.

**Comment by nesnoj** [comment]
> I'd make that a command line parameter without a default. That way one can set it via `egon-data.configuration.yaml` in the working directory and it would be easily available via the dictionary returned by `config.settings()`.

This way we had to tell the users what config they'd have to use to get the same results as in publication XYZ as the yaml is not part of the repo. I'd like to tie the seed to the release. What about the solution in #409?

**Comment by nesnoj** [comment]
PS: There's usually no need to change this param so there's no reason to have it as CLI param.

**Comment by gnn** [comment]
> What about the solution in #409?

I saw that solution but wasn't fast enough with a reply.

> This way we had to tell the users what config they'd have to use to get the same results as in publication XYZ as the yaml is not part of the repo.

Correct. And I would argue that this is the right way of doing it. But that's part of a longer discussion about using `random` which we fortunately don't have to solve right now, because the issue of where to put the seed parameter is orthogonal to how to used it.

> I'd like to tie the seed to the release.

In that case, make it a CLI parameter with a default value. Still easily accessible via `config.settings()`, but with deterministic behaviour unless explicitly modified. Sorry for having my initial suggestion be influenced by my preferences. ;)

**Comment by AmeliaNadal** [comment]
I also don't use random in my scripts.

**Comment by nesnoj** [comment]
#409 has been merged and the CLI param can now be used.
I'll leave this one open until the [other tasks](https://github.com/openego/eGon-data/issues/351#issuecomment-905519619) are done...
